velox/dwio/parquet/reader/CMakeLists.txt

```diff
@@ -36,4 +36,5 @@ target_link_libraries(
   arrow
   Snappy::snappy
   thrift
-  zstd::zstd)
+  zstd::zstd
+  ${Protobuf_LIBRARIES})
```

Comment:
Is this line related and necessary?

Reply:
This is necessary since we are now having references google::protobuf::io::ZeroCopyOutputStream via BufferedOutputStream from dwio/common/compression/Compression.h -> velox/dwio/common/OutputStream.h

Reply:
## undefined reference to `typeinfo for google::protobuf::io::ZeroCopyOutputStream' after this PR

I got the following link issue
```
/usr/bin/ld: velox/dwio/common/compression/libvelox_dwio_common_compression.a(Compression.cpp.o): in function `google::protobuf::io::ZeroCopyOutputStream::ZeroCopyOutputStream()':
/home/chang/SourceCode/velox/cmake-build-debug-system/_deps/protobuf-src/src/google/protobuf/io/zero_copy_stream.h:193: undefined reference to `vtable for google::protobuf::io::ZeroCopyOutputStream'
/usr/bin/ld: velox/dwio/common/compression/libvelox_dwio_common_compression.a(Compression.cpp.o): in function `google::protobuf::io::ZeroCopyOutputStream::~ZeroCopyOutputStream()':
/velox/cmake-build-debug-system/_deps/protobuf-src/src/google/protobuf/io/zero_copy_stream.h:194: undefined reference to `vtable for google::protobuf::io::ZeroCopyOutputStream'
/usr/bin/ld: velox/dwio/common/libvelox_dwio_common.a(OutputStream.cpp.o):(.[data.rel.ro](http://data.rel.ro/)._ZTIN8facebook5velox4dwio6common20BufferedOutputStreamE[_ZTIN8facebook5velox4dwio6common20BufferedOutputStreamE]+0x10): undefined reference to `typeinfo for google::protobuf::io::ZeroCopyOutputStream'
```
Here are my  cmake option

```
-DVELOX_DEPENDENCY_SOURCE=BUNDLED
-DVELOX_ENABLE_EXAMPLES=ON
-DVELOX_ENABLE_PARQUET=ON
-DVELOX_BUILD_TESTING=ON
-DVELOX_ENABLE_BENCHMARKS=ON
-DVELOX_ENABLE_BENCHMARKS_BASIC=ON
```

I checked probobuf cmake  file and make sure **disbale RTTI is OFF**

Not sure how to fix it.

Reply:
We should have fixed the protobuf header leak instead of propagating this dependency. I will open a PR to fix this.